sunpy/net/dataretriever/sources/goes.py

```diff
@@ -52,33 +93,75 @@ def _get_url_for_timerange(self, timerange, **kwargs):
-#       same GOES sat number, which meant we supplied timerange.start and timerange.end to
-#       _get_goes_sat_num. But now, we iterate over all the dates individually and hence
-#       _get_goes_sat_num gets each single date as an argument (the timerange.start and timerange.end
-#       for a single day is the day itself (trivial) ). That is why the same date. Its iterating over all the dates
-#       indivually.
-        for day in all_dates:
-            regex = "{date:%Y}/go{sat:02d}"
-            if (day.end < parse_time('1999/01/15')):
-                regex += "{date:%y%m%d}.fits"
-            else:
-                regex += "{date:%Y%m%d}.fits"
-            url = base_url + regex.format(
-                    date=day.end, sat=self._get_goes_sat_num(day.end, day.end)[0])
-            result.append(url)
+        physobs = kwargs['physobs']
+        def download_particle_flux(time_range):
+            now = datetime.datetime.now()
+            real_time = ((now - time_range.end).total_seconds() <= 300)
+            result = list()
+            total_days = (time_range.end - time_range.start).days + 1
+            all_dates = time_range.split(total_days)
+            url_pattern = 'ftp://ftp.swpc.noaa.gov/pub/lists/particle/{date:%Y%m%d}_G{char}_part_5m.txt'
+            for day in all_dates:
+                result.extend([url_pattern.format(date=day.end, char=ch) for ch in ('p', 's')])
+            if real_time:
+                result.extend(['ftp://ftp.swpc.noaa.gov/pub/lists/particle/G{char}_part_5m.txt'.format(char=ch) for ch in ('p', 's')])
+            return result
+
+        def download_irradiance(time_range):
+            now = datetime.datetime.now()
+            real_time = ((now - time_range.end).total_seconds() <= 300)
+            result = list()
+            total_days = (time_range.end - time_range.start).days + 1
+            all_dates = time_range.split(total_days)
+            url_pattern = 'ftp://ftp.swpc.noaa.gov/pub/lists/xray/{date:%Y%m%d}_G{char}_xr_{cad}m.txt'
+            base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
+            for day in all_dates:
+                regex = "{date:%Y}/go{sat:02d}%Y%m%d.fits"
+                if (now - day.end).days > 3:
+                    url = base_url + regex.format(date=day.end, sat=self._get_goes_sat_num(day.end, day.end)[0])
+                    result.append(url)
+                result.extend([url_pattern.format(char='s', date=day.end, cad=cad) for cad in (1, 5)])
+                result.extend([url_pattern.format(char='p', date=day.end, cad=cad) for cad in (1, 5)])
+            if real_time:
+                result.extend(['ftp://ftp.swpc.noaa.gov/pub/lists/xray/G{char}_xr_1m.txt'.format(char=ch) for ch in ('p', 's')])
+                result.extend(['ftp://ftp.swpc.noaa.gov/pub/lists/xray/G{char}_xr_5m.txt'.format(char=ch) for ch in ('p', 's')])
+            return result
+
+        def download_intensity(time_range):
```

Comment:
This implementation can be now changed with the new suggested changes to the scraper (#1862)